### PR TITLE
Feature: Add back slices

### PR DESCRIPTION
**Pull request by Bilelkihal**
### Done in this PR:

- Add back the slices section to the home page, which indicates the `name (acronym)` of the slice, **the number of ontologies** in this slice, and **the description** of it on hover.
image

- Update the style of the slices notice bar
image

- Make the annotator supports slices by adding the ontologies of the slice by default to the `ontologies` param.
image

- Make the browse page supports slices by filtering the ontologies we get from the endpoint '/submission' to keep only the onces present in '/ontologies' cause it is the endpoint that supports slices in the backend.
image

- Make the mappings page supports slices.
image

### CODE REVIEW

File: app/views/home/index.html.haml [hunk 1]
```diff
@@ -185,7 +185,21 @@
         - $HOME_PAGE_LOGOS&.each do |logo|
           %a{href:logo[:url], target: "_blanc"}
             %img{src: asset_path(logo[:img_src])}
-  
+
+  .home-section
```
Comment (syphax-bouazzouni) [inline app/views/home/index.html.haml:189]: try to put the section in this order: "Slices", "Ontoportals" and "Sponsors"

File: app/views/layouts/_notices.html.haml [hunk 2]
```diff
@@ -6,21 +6,18 @@
...
-        = $SITE
-        = t('layout.notices.slice')
-      = t('layout.notices.at')
-      = succeed "." do
-        %b= request.host
-      = $SITE
-      = succeed "." do
-        %a{:href => "#{$UI_URL}#{request.path}#{query_string}"}
-          = t('layout.notices.visit_the_site')
+  .slice-notice-bar-container
```
Comment (syphax-bouazzouni) [inline app/views/layouts/_notices.html.haml:9]: try to make something, less visible, maybe an icon in the header.
You can't reduce the UX of slices, with that alert.

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by Bilelkihal — COMMENTED** [review]
Reviewed by author

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/controllers/concerns/submission_filter.rb [hunk 3]
```diff
@@ -53,6 +53,7 @@ def submissions_paginate_filter(params)
 
     submissions = sort_submission_by(submissions, @sort_by, @search)
 
+    submissions = submissions.select { |submission| @ontologies.any? { |ontology| ontology.id == submission[:id] } }
```
[the remark this answers is not in the document]
Comment (Bilelkihal) [reply app/controllers/concerns/submission_filter.rb:56]: '/submissions' does not support slices in the back this, but '/ontologies' does, this is why I did this

[the remark this answers is not in the document]
Comment (syphax-bouazzouni) [reply app/controllers/concerns/submission_filter.rb:56]: Yes, but the browse page uses `/ontologies`  that is filled with metadata of `/submissions`, done line 88

[the remark this answers is not in the document]
Comment (Bilelkihal) [reply app/controllers/concerns/submission_filter.rb:56]: In the line 88 we are calling: `LinkedData::Client::Models::OntologySubmission.all` which does not support slices,
image

I should get only 38

[the remark this answers is not in the document]
Comment (syphax-bouazzouni) [reply app/controllers/concerns/submission_filter.rb:56]: you don't have the latest version? it is not the same line as me 
image

[the remark this answers is not in the document]
Comment (Bilelkihal) [reply app/controllers/concerns/submission_filter.rb]: You are right, I just didn't have the latest version of development branch, it's working without the need of filtering them

File: app/views/layouts/_notices.html.haml (hunk 2 above)
Comment (Bilelkihal) [reply app/views/layouts/_notices.html.haml:9]: We'll discuss this tomorrow in the meeting

**Comment by Bilelkihal** [comment]
@jonquet 
Could you confirm to me this description I did using ChatGPT 😅 for the slices so I add it to the languages yml files.

In the end you can see a plus button "create new slice" that will take you to the feedback form, I'm thinking about making a prefilled feedback message.

@syphax-bouazzouni what do you think?

image

**Comment by jonquet** [comment]
Oui tres bien pour l'UI et tres bien pour le bouton "+" mais a ce moment la apelle le "Suggest a new group/slice" 

Pour le message, j'avais mis des choses la : https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/618
Tu peux changer avec ca apres ton "offer a solution!" ... je prefere pas dire que les slices sont par topics. 
Et quand on aura une page pour les groupes et categories on fera un lien sur la partie "and groups" du texte poru aller vers cette page.

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -659,6 +659,11 @@ def cancel_button_component(class_name: nil, id: , value:, data: nil)
     end
   end
  
+  def home_ontoportal_tooltip
```
Comment (syphax-bouazzouni) [inline app/helpers/application_helper.rb:662]: use the config variables

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: config/bioportal_config_env.rb.sample [hunk 5]
```diff
@@ -307,3 +307,6 @@ if $HOSTNAME
     require_relative "bioportal_config_development_#{$HOSTNAME}"
   end
 end
+
+$ontoportal_website_link = "https://ontoportal.org/"
```
Comment (syphax-bouazzouni) [inline config/bioportal_config_env.rb.sample:311]: capital letter for constant, and re-use it in $FOOTER_LINKS

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

**Review by Bilelkihal — COMMENTED** [review]
✅

**Review by syphax-bouazzouni — APPROVED** [review]
Approved the slices part, but waiting in another PR for the homepage help texts

**Pull request merged**